client/layout/masterbar/quick-language-switcher.jsx

```diff
@@ -0,0 +1,59 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { getLocaleSlug } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+
+import MasterbarItem from './item';
+import LanguagePickerModal from 'components/language-picker/modal';
+import switchLocale from 'lib/i18n-utils/switch-locale';
+import config from 'config';
+
+class QuickLanguageSwitcher extends React.Component {
+	state = {
+		isShowingModal: false,
+	};
+
+	toggleLanguagesModal = () => {
+		this.setState( state => ( {
+			isShowingModal: ! state.isShowingModal,
+		} ) );
+	};
+
+	onClick = event => {
+		this.toggleLanguagesModal();
+		event.preventDefault();
+		return;
+	};
+
+	onSelected = languageSlug => {
+		switchLocale( languageSlug );
+	};
+
+	render() {
+		const selectedLanguageSlug = getLocaleSlug();
+
+		return (
+			<div className="masterbar__item quick-language-switcher">
+				<MasterbarItem icon="globe" onClick={ this.onClick }>
+					{ getLocaleSlug() }
```

Comment:
We've already assigned `getLocaleSlug()` to `selectedLanguageSlug`. Can we use `selectedLanguageSlug` here instead?

Reply:
Oops, that's what I had intended with introducing `selectedLanguageSlug`.